src/main/java/net/sf/jabref/gui/search/SearchWorker.java

```diff
@@ -60,44 +60,52 @@ protected void done() {
     }
 
     private void updateUIWithSearchResult(List<BibEntry> matchedEntries) {
+        GlobalSearchBar globalSearchBar = JabRefGUI.getMainFrame().getGlobalSearchBar();
 
         // check if still the current query
-        if (!basePanel.getSearchBar().isStillValidQuery(searchQuery)) {
+        if (!globalSearchBar.isStillValidQuery(searchQuery)) {
             // do not update - another search was already issued
             return;
         }
 
-        // clear
-        for (BibEntry entry : basePanel.getDatabase().getEntries()) {
-            entry.setSearchHit(false);
-        }
-
-        for (BibEntry entry : matchedEntries) {
-            entry.setSearchHit(true);
-        }
+        basePanel.getDatabase().getEntries().forEach(entry -> entry.setSearchHit(false));
```

Comment:
This applies only to the for each, because it's harder to debug. We had a discussion about it once. @koppor